src/Type/ArrayType.php

```diff
@@ -367,6 +367,17 @@ public function unsetOffset(Type $offsetType): Type
 		return $this;
 	}
 
+	public function fillKeysArray(Type $valueType): Type
+	{
+		$itemType = $this->getIterableValueType();
+
+		if ((new IntegerType())->isSuperTypeOf($itemType)->no() && !$itemType->toString() instanceof ErrorType) {
+			return new ArrayType($itemType->toString(), $valueType);
+		}
```

Comment:
not sure how heavy `->toString()` is, but we could assign it to a var here and in `ConstantArrayType` maybe to avoid calling it twice

Reply:
adapted in the second commit